databroker/broker.py

```diff
@@ -1060,3 +1060,16 @@ def inner(stream):
                     yield 'stop', new_stop
         return inner
     return outer
+
+
+def header_io(db_in, db_out):
+    def outer(f):
+        def inner(header):
+            output_uids = []
+            stream = db_in.restream(header, fill=True)
```

Comment:
Would it be possible to do this in a many header/stream case 
Streams = [restream for hdr in hdrs]